pkg/client/record/event.go

```diff
@@ -111,19 +112,19 @@ func (eventBroadcaster *eventBroadcasterImpl) StartRecordingToSink(sink EventSin
 			// Events are safe to copy like this.
 			eventCopy := *event
 			event = &eventCopy
-
+			var patch []byte
 			previousEvent := eventCache.getEvent(event)
 			updateExistingEvent := previousEvent.Count > 0
 			if updateExistingEvent {
-				event.Count = previousEvent.Count + 1
-				event.FirstTimestamp = previousEvent.FirstTimestamp
+				// we still need to copy Name because the Patch relies on the Name to find the target event
 				event.Name = previousEvent.Name
-				event.ResourceVersion = previousEvent.ResourceVersion
+				time, _ := event.LastTimestamp.MarshalJSON()
+				patch = []byte(fmt.Sprintf("{\"count\":%d,\"lastTimestamp\":%s}", previousEvent.Count+1, string(time)))
```

Comment:
Is this the correct way to do a patch? It feels weird that the client has to construct and pass in serialized json bytes. 

If so, is there any common util method for constructing a patch, that you could use instead?